inductor(CPU): add Conv+binary+unary fusion filter

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90259


For Conv+binary+unary fusion, we only support conv+add+relu, this PR adds a such check to fix TIMM failed models. 
TODO: enable more Conv+binary+unary fusion to improve TIMM models' performance.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
inductor(CPU): add Conv+binary+unary fusion filter

[ghstack-poisoned]

EikanWang commented on torch/_inductor/overrides.py:
```diff
@@ -718,6 +718,12 @@ def fuse_unary(gm: torch.fx.GraphModule):
+                if type(computation_node) in [
+                    ConvBinary2d,
+                    ConvBinaryInplace2d,
+                ] and type(unary_node) not in [nn.ReLU]:
+                    continue
```
What's the behavior of `F.relu`?

XiaobingSuper replied:
We don't support it now, only support **nn.module** fusion, for function path, we can enable it at the next step.

EikanWang replied:
@XiaobingSuper , I'm okay with this PR to fix TIMM failed model. And let's submit another PR to support `F.ReLU`.

EikanWang reviewed (commented):
LGTM

XiaobingSuper committed:
Update on "inductor(CPU): add Conv+binary+unary fusion filter"



For Conv+binary+unary fusion, we only support conv+add+relu, this PR adds a such check to fix TIMM failed models. 
TODO: enable more Conv+binary+unary fusion to improve TIMM models' performance.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@jansel

XiaobingSuper commented:
@pytorchbot rebase

XiaobingSuper committed:
Update on "inductor(CPU): add Conv+binary+unary fusion filter"



For Conv+binary+unary fusion, we only support conv+add+relu, this PR adds a such check to fix TIMM failed models. 
TODO: enable more Conv+binary+unary fusion to improve TIMM models' performance.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]
